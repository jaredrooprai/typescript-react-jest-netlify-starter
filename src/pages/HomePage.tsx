import React from 'react';
import NavBar from '@components/NavBar';
import { useQuery } from 'react-query';
import axios, { AxiosResponse } from 'axios';

const HomePage: React.FC = () => {
  const { data } = useQuery('user', (): Promise<AxiosResponse> => axios.get('api/auth/profile'));
  return (
    <>
      <NavBar />
      <h2>Home Page</h2>
      {JSON.stringify(data)}
    </>
  );
};
export default HomePage;
