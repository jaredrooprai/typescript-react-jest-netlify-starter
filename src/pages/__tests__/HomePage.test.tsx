import React from 'react';
import HomePage from '@pages/HomePage';
import { fireEvent, render } from '@utils/test-utils';
import axios from 'axios';

test('has title', () => {
  const { debug, getByText } = render(<HomePage />);
  fireEvent.click(getByText('Load User'));

  debug();
  expect(getByText('Home Page')).toBeInTheDocument;
});
