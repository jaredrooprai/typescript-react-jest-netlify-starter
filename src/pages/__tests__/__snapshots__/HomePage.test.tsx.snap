// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`should take a snapshot 1`] = `
<DocumentFragment>
  <h2>
    User Credit Cards
  </h2>
  <button>
    Load Credit Cards
  </button>
  <div
    style="height: 40px;"
  />
</DocumentFragment>
`;
